torch/csrc/jit/codegen/cuda/interface.cpp

```diff
@@ -142,8 +142,8 @@ bool complyWith(
                 stride_properties[j - 1]->stride_index_.has_value(),
                 "Counknown index is meaningless");
             // TODO: merge this check up
-            if (t_strides[sorted_index] !=
-                t_strides[inner_dim] * t_sizes[inner_dim]) {
+            if ((t_strides[sorted_index] != 0) and (t_strides[sorted_index] !=
```

Comment:
`and`?

Reply:
So this ended up being that a dimension != 1, but was broadcasted by setting stride = 0? I see how this check would have indicated it was contiguous even though it was really broadcasted.

Reply:
This is not safe any more is it?

If we mark a broadcasted dimension as contiguous, we might ended up collapsing the dimension to its inner dim, then we'll be indexing into broadcasted dimension and that's gonna be IMA.